### PR TITLE
{data}[intel/2015b] DBD-mysql 4.032 (REVIEW)

### PR DESCRIPTION
**Pull request by boegel**
requires #1991

### CODE REVIEW

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 4 out of 4 (4 easyconfigs in this PR)
Linux SL 6.6, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.6
See https://gist.github.com/6191e6fcab4af430e169 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 4 out of 4 (4 easyconfigs in this PR)
Linux centos linux 7.1.1503, Intel(R) Xeon(R) CPU E5-2680 v3 @ 2.50GHz, Python 2.7.5
See https://gist.github.com/ba7898c9db1cff7b71bd for a full test report.

**Comment by boegel** [comment]
Jenkins: test this please

**Comment by boegel** [comment]
@wpoely86: please review(/test)?

File: easybuild/easyconfigs/m/MySQL/MySQL-5.6.26-intel-2015b-clientonly.eb [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+ open-source relational database management system (RDBMS)."""
+
+source_urls = ['http://dev.mysql.com/get/Downloads/MySQL-%(version_major_minor)s/']
+sources = [SOURCELOWER_TAR_GZ]
+
+toolchain = {'name': 'intel', 'version': '2015b'}
+
+dependencies = [
+    ('libevent', '2.0.22'),
+    ('libreadline', '6.3', '', ('GNU', '4.9.3-2.25')),
```
Comment (ehiggs) [inline easybuild/easyconfigs/m/MySQL/MySQL-5.6.26-intel-2015b-clientonly.eb]: I don't see [an EasyBuild config for libreadline-6.3-GCC-4.9.3](https://github.com/hpcugent/easybuild-easyconfigs/tree/master/easybuild/easyconfigs/l/libreadline). Should this be in the PR?

Comment (boegel) [inline easybuild/easyconfigs/m/MySQL/MySQL-5.6.26-intel-2015b-clientonly.eb]: It's in develop already.

File: easybuild/easyconfigs/d/DBD-mysql/DBD-mysql-4.032-intel-2015b-Perl-5.20.3.eb [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+source_urls = ['http://cpan.metacpan.org/authors/id/C/CA/CAPTTOFU/']
+sources = [SOURCE_TAR_GZ]
+
+perl = 'Perl'
+perlver = '5.20.3'
+versionsuffix = '-%s-%s' % (perl, perlver)
+
+dependencies = [
+    (perl, perlver),
+    ('MySQL', '5.6.26', '-clientonly'),
```
Comment (wpoely86) [inline easybuild/easyconfigs/d/DBD-mysql/DBD-mysql-4.032-intel-2015b-Perl-5.20.3.eb]: `MySQL` is so old school. The cool people of today use `MariaDB`.

File: easybuild/easyconfigs/l/libevent/libevent-2.0.22-intel-2015b.eb [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+homepage = 'http://libevent.org/'
+description = """The libevent API provides a mechanism to execute a callback function when a specific
+ event occurs on a file descriptor or after a timeout has been reached.
+ Furthermore, libevent also support callbacks due to signals or regular timeouts."""
+
+toolchain = {'name': 'intel', 'version': '2015b'}
+
+source_urls = [
+    'https://sourceforge.net/projects/levent/files/%(name)s/%(name)s-%(version_major_minor)s/',
+    'https://github.com/downloads/%(name)s/%(name)s/',
```
Comment (wpoely86) [inline easybuild/easyconfigs/l/libevent/libevent-2.0.22-intel-2015b.eb]: I would reverse the order. Sourceforge is not to be trusted anymore.

File: easybuild/easyconfigs/m/MySQL/MySQL-5.6.26-intel-2015b-clientonly.eb [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+versionsuffix = '-clientonly'
+
+homepage = 'http://www.mysql.com/'
+description = """MySQL is (as of March 2014) the world's second most widely used
+ open-source relational database management system (RDBMS)."""
+
+source_urls = ['http://dev.mysql.com/get/Downloads/MySQL-%(version_major_minor)s/']
+sources = [SOURCELOWER_TAR_GZ]
+
+toolchain = {'name': 'intel', 'version': '2015b'}
```
Comment (wpoely86) [inline easybuild/easyconfigs/m/MySQL/MySQL-5.6.26-intel-2015b-clientonly.eb]: why not build it with `GNU`?

Comment (wpoely86) [inline easybuild/easyconfigs/m/MySQL/MySQL-5.6.26-intel-2015b-clientonly.eb]: or `iccifort` ?

Comment (boegel) [inline easybuild/easyconfigs/m/MySQL/MySQL-5.6.26-intel-2015b-clientonly.eb]: I guess I could, yeah.

File: easybuild/easyconfigs/l/libevent/libevent-2.0.22-intel-2015b.eb (hunk 3 above)
Comment (boegel) [inline easybuild/easyconfigs/l/libevent/libevent-2.0.22-intel-2015b.eb]: ok

File: easybuild/easyconfigs/d/DBD-mysql/DBD-mysql-4.032-intel-2015b-Perl-5.20.3.eb (hunk 2 above)
Comment (boegel) [inline easybuild/easyconfigs/d/DBD-mysql/DBD-mysql-4.032-intel-2015b-Perl-5.20.3.eb]: not trivial to switch to MariaDB (but we're getting there), see #293

**Comment by ehiggs** [comment]
Looks good. I was able to install it and load the module.

File: easybuild/easyconfigs/d/DBD-mysql/DBD-mysql-4.032-intel-2015b-Perl-5.20.3.eb [hunk 5]
```diff
@@ -0,0 +1,31 @@
+easyblock = 'PerlModule'
+
+name = 'DBD-mysql'
+version = '4.032'
+
+homepage = 'http://search.cpan.org/~capttofu/DBD-mysql-%(version)s/'
+description = """Perl binding for MySQL"""
+
+toolchain = {'name': 'intel', 'version': '2015b'}
```
Comment (wpoely86) [inline easybuild/easyconfigs/d/DBD-mysql/DBD-mysql-4.032-intel-2015b-Perl-5.20.3.eb]: is there any point of doing this with intel instead of iccifort ?

**Comment by wpoely86** [comment]
lgtm

File: easybuild/easyconfigs/d/DBD-mysql/DBD-mysql-4.032-intel-2015b-Perl-5.20.3.eb (hunk 5 above)
Comment (boegel) [inline easybuild/easyconfigs/d/DBD-mysql/DBD-mysql-4.032-intel-2015b-Perl-5.20.3.eb]: I'm not going to start installing Perl modules with subtoolchains (at least not until @ocaisa's subtoolchain stuff is working)

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 5 out of 5 (5 easyconfigs in this PR)
Linux centos linux 7.1.1503, Intel(R) Xeon(R) CPU E5-2680 v3 @ 2.50GHz, Python 2.7.5
See https://gist.github.com/d03967577a6063b10401 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 5 out of 5 (5 easyconfigs in this PR)
Linux SL 6.7, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.6
See https://gist.github.com/2f5c816ada2583fbd371 for a full test report.

**Comment by boegel** [comment]
Thanks for the review @wpoely86 and @ehiggs!

**Pull request merged**